### DISCUSSION TITLE
Cloud notifications with image url, image is not showing on some iOS devices

### DISCUSSION THREAD

**Discussion by Ggayane** [invertase/react-native-firebase#6313] [Q&A] [answered]
I've implemented Notification image service as it's documented [here](https://rnfirebase.io/messaging/ios-notification-images#step-1---add-a-notification-service-extension) and when I send a notification with image (size of ~150kb) on some iOS devices image is not visible. 

Does anyone know what can be the reason?

Here are the screenshots of service extension:

<img width="1372" alt="Screen Shot 2022-06-16 at 13 40 23" src="https://user-images.githubusercontent.com/15812978/174042132-86d190f4-bd16-4754-88e8-4c446cae629b.png">

<img width="922" alt="Screen Shot 2022-06-16 at 13 39 47" src="https://user-images.githubusercontent.com/15812978/174042159-d0a63b64-db5d-4804-b73b-92a1242ee667.png">

<img width="731" alt="Screen Shot 2022-06-16 at 13 38 46" src="https://user-images.githubusercontent.com/15812978/174042166-ea4bc3fb-c301-4932-b6e9-d565fd6a2a8d.png">

**Comment by mikehardy** [comment, marked as answer]
As far as I know, it's a best effort. Notification post-receipt / pre-notification-center-post work has a fixed time slice to complete all work (including download of image) and if it does not complete in time, that's it. Incomplete work is terminated and notification is posted without alteration.

**Reply by Ggayane** [reply]
thank you for your answer!
